### DISCUSSION TITLE
install packages in the paddlepaddle env

### DISCUSSION THREAD

**Discussion by begovegar** [PaddlePaddle/PaddleSpatial#165] [BaiduKDDCup2022]
Could you please install the following packages in the paddlepaddle enviroment?
pip install skforecast
pip install prophet  
pip install sktime
pip install statsmodels

We use all of them to forecast all the variables unknown.
Thank you.

**Comment by ramber1836** [comment]
These packages have been added to the base environment.
I cannot see any necessity of installing these packages in the paddlepaddle environment.

**Comment by begovegar** [comment]
Hi, we are going to estimate some of the missing values with this packages, variables that will enter as input in the Neural network trained in the paddlepaddle enviroment.  This is the reason why we ask you to include them. Maybe I didn't explain it very well. 
Do you think you can include them now?
Thanks!

**Reply by ramber1836** [reply]
We will let you know when it is ready.

**Reply by ramber1836** [reply]
Tried to install the prophet and related packages in the PaddlePaddle environment, but they are incompatible with the current installation of the PaddlePaddle environment: 

Building wheels for collected packages: prophet
  Building wheel for prophet (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /usr/local/bin/conda/envs/paddlepaddle-env/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-wc7vh4sk/prophet_d2c8016d999a48bbbc733eb15d09226c/setup.py'"'"'; __file__='"'"'/tmp/pip-install-wc7vh4sk/prophet_d2c8016d999a48bbbc733eb15d09226c/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-kw3k1tte
       cwd: /tmp/pip-install-wc7vh4sk/prophet_d2c8016d999a48bbbc733eb15d09226c/
  Complete output (52 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib
  creating build/lib/prophet
  creating build/lib/prophet/stan_model
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/tmp/pip-install-wc7vh4sk/prophet_d2c8016d999a48bbbc733eb15d09226c/setup.py", line 150, in <module>
      long_description_content_type='text/markdown',
    File "/usr/local/bin/conda/envs/paddlepaddle-env/lib/python3.7/site-packages/setuptools/__init__.py", line 87, in setup
      return distutils.core.setup(**attrs)
    File "/usr/local/bin/conda/envs/paddlepaddle-env/lib/python3.7/site-packages/setuptools/_distutils/core.py", line 148, in setup
      return run_commands(dist)
    File "/usr/local/bin/conda/envs/paddlepaddle-env/lib/python3.7/site-packages/setuptools/_distutils/core.py", line 163, in run_commands
      dist.run_commands()
    File "/usr/local/bin/conda/envs/paddlepaddle-env/lib/python3.7/site-packages/setuptools/_distutils/dist.py", line 967, in run_commands
      self.run_command(cmd)
    File "/usr/local/bin/conda/envs/paddlepaddle-env/lib/python3.7/site-packages/setuptools/dist.py", line 1214, in run_command
      super().run_command(command)
    File "/usr/local/bin/conda/envs/paddlepaddle-env/lib/python3.7/site-packages/setuptools/_distutils/dist.py", line 986, in run_command
      cmd_obj.run()
    File "/usr/local/bin/conda/envs/paddlepaddle-env/lib/python3.7/site-packages/wheel/bdist_wheel.py", line 299, in run
      self.run_command('build')
    File "/usr/local/bin/conda/envs/paddlepaddle-env/lib/python3.7/site-packages/setuptools/_distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/usr/local/bin/conda/envs/paddlepaddle-env/lib/python3.7/site-packages/setuptools/dist.py", line 1214, in run_command
      super().run_command(command)
    File "/usr/local/bin/conda/envs/paddlepaddle-env/lib/python3.7/site-packages/setuptools/_distutils/dist.py", line 986, in run_command
      cmd_obj.run()
    File "/usr/local/bin/conda/envs/paddlepaddle-env/lib/python3.7/site-packages/setuptools/_distutils/command/build.py", line 135, in run
      self.run_command(cmd_name)
    File "/usr/local/bin/conda/envs/paddlepaddle-env/lib/python3.7/site-packages/setuptools/_distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/usr/local/bin/conda/envs/paddlepaddle-env/lib/python3.7/site-packages/setuptools/dist.py", line 1214, in run_command
      super().run_command(command)
    File "/usr/local/bin/conda/envs/paddlepaddle-env/lib/python3.7/site-packages/setuptools/_distutils/dist.py", line 986, in run_command
      cmd_obj.run()
    File "/tmp/pip-install-wc7vh4sk/prophet_d2c8016d999a48bbbc733eb15d09226c/setup.py", line 48, in run
      build_models(target_dir)
    File "/tmp/pip-install-wc7vh4sk/prophet_d2c8016d999a48bbbc733eb15d09226c/setup.py", line 36, in build_models
      from prophet.models import StanBackendEnum
    File "/tmp/pip-install-wc7vh4sk/prophet_d2c8016d999a48bbbc733eb15d09226c/prophet/__init__.py", line 8, in <module>
      from prophet.forecaster import Prophet
    File "/tmp/pip-install-wc7vh4sk/prophet_d2c8016d999a48bbbc733eb15d09226c/prophet/forecaster.py", line 17, in <module>
      from prophet.make_holidays import get_holiday_names, make_holidays_df
    File "/tmp/pip-install-wc7vh4sk/prophet_d2c8016d999a48bbbc733eb15d09226c/prophet/make_holidays.py", line 14, in <module>
      import prophet.hdays as hdays_part2
    File "/tmp/pip-install-wc7vh4sk/prophet_d2c8016d999a48bbbc733eb15d09226c/prophet/hdays.py", line 13, in <module>
      from convertdate.islamic import from_gregorian, to_gregorian
  ModuleNotFoundError: No module named 'convertdate'
  ----------------------------------------
  ERROR: Failed building wheel for prophet
  Running setup.py clean for prophet
Failed to build prophet
Installing collected packages: pymeeus, ujson, korean-lunar-calendar, hijri-converter, ephem, convertdate, tqdm, setuptools-git, pystan, LunarCalendar, holidays, cmdstanpy, prophet
    Running setup.py install for prophet ... |

To ensure the stableness of the PaddlePaddle environment in the online test, these package requirements cannot be satisfied. 
Apologies for that.

**Reply by sergiosmg96** [reply]
We are sorry to know about that. Does that mean that the other packages (skforecast, sktime, statsmodels) were not installed as well? Thank you for your attention. Kind regards

**Reply by ramber1836** [reply]
When installing the others, similar problems are encountered. We are working on it and will let you know if it works.

**Comment by begovegar** [comment]
Ok, thanks for trying.
We installed the packages in our paddlepaddle environment without problems.

[https://www.innova-tsn.com/wp-content/uploads/2020/02/innova-tsn-CMYK-vectorial_transparente.png]<https://www.innova-tsn.com/>

Begoña Vega
 Senior Consultant
Advanced Analytics & Big Data
    MADRID - BARCELONA - LONDRES
                  CDMX - BOGOTÁ

                  www.innova-tsn.com


 Plaza Manuel Gómez Moreno 2, Plantas 15º y 17ºB
 28020 ***@***.***,-3.696314,17z/data=!3m1!4b1!4m5!3m4!1s0xd4228fb6e9fed9f:0x4415e616cfb0712!8m2!3d40.44863!4d-3.69412>

[Twitter]<https://twitter.com/innovatsn>[LinkedIn]<https://www.linkedin.com/company/innovatsn/?originalSubdomain=es>[YouTube]<https://www.youtube.com/channel/UCnhhzy6vkYEtSoRSwMAq54Q>
 Tel. +34915130073
CONFIDENCIALIDAD La información contenida en este correo electrónico es confidencial/privilegiada y está destinada a ser leída sólo por la(s) persona (s a la(s) que va dirigida. Si usted lee este documento y no es el destinatario señalado, o ha recibido esta comunicación por error, le informamos de que está totalmente prohibida y puede ser ilegal cualquier divulgación, distribución o reproducción de esta comunicación, le rogamos que nos lo notifique inmediatamente elimine dicho correo de su equipo. En el caso de ser el destinatario del mismo no podrá usarlo para fines distintos a los de la propia comunicación que contiene, ni tampoco ser utilizado en juicio (artículo 34.e) de RD 58/2001 de 22 de abril, por el que se aprueba el Estatuto General de la Abogacía Española).

De: Xinjiang Lu ***@***.***>
Enviado el: martes, 14 de junio de 2022 13:53
Para: PaddlePaddle/PaddleSpatial ***@***.***>
CC: Begoña Vega ***@***.***>; Author ***@***.***>
Asunto: Re: [PaddlePaddle/PaddleSpatial] install packages in the paddlepaddle env (Discussion #165)


When installing the others, similar problems are encountered. We are working on it and will let you know if it works.

**Reply by begovegar** [reply]
Ok, thank you!! We are looking fordward for your news.

**Reply by ramber1836** [reply]
Currently installed pgl (2.2.3) requires numpy<1.20.0,>=1.16.4, but sktime 0.11.4 requires numpy<1.22,>=1.21.0, and skforecast 0.4.3 requires numpy<=1.22,>=1.20, which are incompatible.

**Comment by begovegar** [comment]
Hi, so if we only install sktime there is no issue, right? Prioritize sktime in this case.
Thanks!

De: Xinjiang Lu ***@***.***>
Enviado el: miércoles, 15 de junio de 2022 8:41
Para: PaddlePaddle/PaddleSpatial ***@***.***>
CC: Begoña Vega ***@***.***>; Author ***@***.***>
Asunto: Re: [PaddlePaddle/PaddleSpatial] install packages in the paddlepaddle env (Discussion #165)


Currently installed pgl (2.2.3) requires numpy<1.20.0,>=1.16.4, but sktime 0.11.4 requires numpy<1.22,>=1.21.0, and skforecast 0.4.3 requires numpy<=1.22,>=1.20, which are in compatible.

**Reply by ramber1836** [reply]
No, sktime is incompatible with the installed pgl package.

**Comment by begovegar** [comment]
Ok so **statsmodel** was installed then? Is it possible to install also **math**?

**Reply by ramber1836** [reply]
Please provide the complete information about the package.

**Comment by begovegar** [comment]
Which package you mean? Statsmodel or math? Complete information about the purpose you mean?

**Reply by ramber1836** [reply]
The math package.

**Comment by begovegar** [comment]
We want to make some calculations with distances and angles...to imput some missing values

**Reply by ramber1836** [reply]
Please provide the complete pip install command for the math package. 
The math package you mentioned, is the built-in math package in the standard Python library? If so, pip install is not necessary.

**Comment by begovegar** [comment]
Yes, I think it should be so probably pip install is not needed.
Thanks!

[https://www.innova-tsn.com/wp-content/uploads/2020/02/innova-tsn-CMYK-vectorial_transparente.png]<https://www.innova-tsn.com/>

Begoña Vega
 Senior Consultant
Advanced Analytics & Big Data
    MADRID - BARCELONA - LONDRES
                  CDMX - BOGOTÁ

                  www.innova-tsn.com


 Plaza Manuel Gómez Moreno 2, Plantas 15º y 17ºB
 28020 ***@***.***,-3.696314,17z/data=!3m1!4b1!4m5!3m4!1s0xd4228fb6e9fed9f:0x4415e616cfb0712!8m2!3d40.44863!4d-3.69412>

[Twitter]<https://twitter.com/innovatsn>[LinkedIn]<https://www.linkedin.com/company/innovatsn/?originalSubdomain=es>[YouTube]<https://www.youtube.com/channel/UCnhhzy6vkYEtSoRSwMAq54Q>
 Tel. +34915130073
CONFIDENCIALIDAD La información contenida en este correo electrónico es confidencial/privilegiada y está destinada a ser leída sólo por la(s) persona (s a la(s) que va dirigida. Si usted lee este documento y no es el destinatario señalado, o ha recibido esta comunicación por error, le informamos de que está totalmente prohibida y puede ser ilegal cualquier divulgación, distribución o reproducción de esta comunicación, le rogamos que nos lo notifique inmediatamente elimine dicho correo de su equipo. En el caso de ser el destinatario del mismo no podrá usarlo para fines distintos a los de la propia comunicación que contiene, ni tampoco ser utilizado en juicio (artículo 34.e) de RD 58/2001 de 22 de abril, por el que se aprueba el Estatuto General de la Abogacía Española).

De: Xinjiang Lu ***@***.***>
Enviado el: miércoles, 15 de junio de 2022 15:55
Para: PaddlePaddle/PaddleSpatial ***@***.***>
CC: Begoña Vega ***@***.***>; Author ***@***.***>
Asunto: Re: [PaddlePaddle/PaddleSpatial] install packages in the paddlepaddle env (Discussion #165)


Please provide the complete pip install command for the math package.
The math package you mentioned, is the built-in math package in the standard Python library? If so, pip install is not necessary.
